src/core/transaction.c

```diff
@@ -628,7 +628,9 @@ static int transaction_apply(
                 HASHMAP_FOREACH(j, m->jobs) {
                         assert(j->installed);
 
-                        if (j->unit->ignore_on_isolate)
+                        if (j->unit->ignore_on_isolate ||
+                                        (manager_state(j->unit->manager) != MANAGER_RUNNING &&
```

Comment:
I'm not sure if checking `manager_state` is a reliable way for detecting `soft-reboot`? manager can also be still starting or in `MANAGER_DEGRADED`. Maybe check the unit we're isolating to instead?

Reply:
When isolating on boot we don't know what the unit will be, as it can be specified for example on the cmdline with systemd.unit. Starting/initializing/stopping is a good approximation, as there are other mechanisms that will stop the unit anyway on real shutdown.